### PR TITLE
Support nn.EmbeddingBag

### PR DESCRIPTION
**Pull request by alexandresablayrolles**
WIP. Passing activations as a list instead of as a tensor.

### CODE REVIEW

File: opacus/grad_sample/embedding.py [hunk 1]
```diff
@@ -53,3 +54,25 @@ def compute_embedding_grad_sample(
         torch.backends.cudnn.deterministic = saved
         ret[layer.weight] = grad_sample
     return ret
+
+
+@register_grad_sampler(nn.EmbeddingBag)
+def compute_embeddingbag_gradsampler(layer, inputs, backprops):
```
Comment (ffuuugor) [inline opacus/grad_sample/embedding.py:60]: Can we maybe have a test for it, like we have for other supported layers?

**Review by ffuuugor — CHANGES_REQUESTED** [review]
Cool, definitely make sense given that the inputs to the backward hook itself is a list of tensors.
Left one inline comment. Additionally, please take a look at the linter output. Unit test error is irrelevant as it's inherited from main, but the flake is new I think

**Review by ffuuugor — APPROVED** [review]

File: opacus/grad_sample/grad_sample_module.py [hunk 2]
```diff
@@ -282,7 +282,7 @@ def capture_activations_hook(
 
         if not hasattr(module, "activations"):
             module.activations = []
-        module.activations.append(forward_input[0].detach())  # pyre-ignore
+        module.activations.append([t.detach() for t in forward_input])  # pyre-ignore
```
Comment (karthikprasad) [inline opacus/grad_sample/grad_sample_module.py:285]: should this be append or extend?

Comment (ffuuugor) [reply opacus/grad_sample/grad_sample_module.py:285]: I think append is right, as activations is a list of tuples in this case

**Comment by ffuuugor** [comment]
LGTM

**Pull request closed without merging**